# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1551132352 0
#      Mon Feb 25 22:05:52 2019 +0000
# Node ID 0b17abf0ee9d03b5e91f31489ed4b62048638cac
# Parent  7e48abcc10322899aceed17158999748cebd86a1
fix error introduced in changeset 287eba9ed14b


* oct-parse.yy (base_parser::start_function): Reverse sense of test
added in previous change.

diff --git a/libinterp/parse-tree/oct-parse.yy b/libinterp/parse-tree/oct-parse.yy
--- a/libinterp/parse-tree/oct-parse.yy
+++ b/libinterp/parse-tree/oct-parse.yy
@@ -3398,26 +3398,26 @@ namespace octave
         if (m_fcn_file_from_relative_lookup)
           fcn->mark_relative ();
 
         if (m_lexer.m_parsing_class_method)
           {
             if (m_lexer.m_parsing_classdef)
               {
                 if (m_curr_class_name == id_name)
-                  fcn->mark_as_legacy_constructor ();
+                  fcn->mark_as_classdef_constructor ();
                 else
-                  fcn->mark_as_legacy_method ();
+                  fcn->mark_as_classdef_method ();
               }
             else
               {
                 if (m_curr_class_name == id_name)
-                  fcn->mark_as_classdef_constructor ();
+                  fcn->mark_as_legacy_constructor ();
                 else
-                  fcn->mark_as_classdef_method ();
+                  fcn->mark_as_legacy_method ();
               }
 
             fcn->stash_dispatch_class (m_curr_class_name);
           }
 
         std::string nm = fcn->fcn_file_name ();
 
         sys::file_stat fs (nm);
